calico-vpp-agent/watchers/peers_watcher.go

```diff
@@ -21,25 +21,34 @@ import (
 
 	bgpapi "github.com/osrg/gobgp/api"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	tomb "gopkg.in/tomb.v2"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
 	calicov3 "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
 	oldv3 "github.com/projectcalico/calico/libcalico-go/lib/apis/v3"
 	"github.com/projectcalico/calico/libcalico-go/lib/backend/api"
 	calicov3cli "github.com/projectcalico/calico/libcalico-go/lib/clientv3"
 	"github.com/projectcalico/calico/libcalico-go/lib/options"
 	"github.com/projectcalico/calico/libcalico-go/lib/selector"
 	"github.com/projectcalico/calico/libcalico-go/lib/watch"
+
 	"github.com/projectcalico/vpp-dataplane/calico-vpp-agent/common"
 	"github.com/projectcalico/vpp-dataplane/config/config"
-	"github.com/sirupsen/logrus"
-	"golang.org/x/net/context"
-	tomb "gopkg.in/tomb.v2"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type PeerWatcher struct {
 	log      *logrus.Entry
 	clientv3 calicov3cli.Interface
 
+	// Subcomponent for accessing and watching secrets (that hold BGP passwords).
+	secretWatcher          *secretWatcher
+	secretWatcherEventChan chan string
```

Comment:
We can probably use the `peerWatcherEventChan` instead of creating a new chan for secrets